crates/oxc_linter/src/rules/eslint/max_len.rs

```diff
@@ -0,0 +1,781 @@
+                        // need to remove the len of "/*"
+                        is_trailing = span.start - COMMENT_LENGTH > line_span.start
+                            && !source_text
+                                [line_span.start as usize..(span.start - COMMENT_LENGTH) as usize]
+                                .trim()
+                                .is_empty();
+
+                        is_full_line_comment = !is_trailing;
+                    }
+
+                    if comment_end_line == line_index {
+                        // the end of a multi-line comment
+                        if line_span.end == span.end + COMMENT_LENGTH {
+                            is_full_line_comment = true;
+                            is_trailing = false;
+                        }
+                    }
+
+                    // the middle of a multi-line comment
+                    if comment_start_line < line_index && comment_end_line > line_index {
+                        is_full_line_comment = true;
+                    }
+                }
+                _ => (),
+            }
+        }
+
+        (has_comment, is_trailing, is_full_line_comment)
+    }
+
+    fn compute_line_length(line: &str, tab_width: usize) -> usize {
+        let mut extra_character_count: isize = 0;
+        for cap in TAB_REGEXP.captures_iter(line) {
+            if let Some(match_) = cap.get(0) {
+                let total_offset = match_.start() as isize + extra_character_count;
+                let previous_tab_stop_offset =
+                    if tab_width != 0 { total_offset % tab_width as isize } else { 0 };
+
+                let space_count = tab_width as isize - previous_tab_stop_offset;
+                extra_character_count += space_count - 1; // -1 for the replaced tab
+            }
+        }
+
+        (line.chars().count() as isize + extra_character_count) as usize
+    }
+
+    fn check_is_in_line(strings_or_literals: &Vec<Span>, line_span: &Span) -> bool {
+        strings_or_literals.iter().any(|span| {
+            (span.start >= line_span.start && line_span.end >= span.start) // in start
+                || (span.end >= line_span.start && line_span.end >= span.end) // in end
+                || (span.start <= line_span.start && line_span.end <= span.end) // in middle or single line
+        })
+    }
+
+    fn should_ignore_line(
+        text_to_measure: &str,
+        ignore_pattern_re: &Option<Regex>,
+        ignore_urls: bool,
+        ignore_strings: bool,
+        ignore_template_literals: bool,
+        ignore_reg_exp_literals: bool,
+        literal_spans: &LiteralSpans,
+        line_span: &Span,
+    ) -> bool {
+        if let Some(ignore_pattern_re) = ignore_pattern_re {
+            if ignore_pattern_re.is_match(text_to_measure) {
+                return true;
+            }
+        }
+
+        if ignore_urls && URL_REGEXP.is_match(text_to_measure) {
```

Comment:
> This PR has a performance issue now, it may be caused by using `Parser`, you can try to remove this if it still exists.
> 
> > Avoid the regex crate when possible. Regexes are slow, most of them can be rewritten in a performant way by using Rust iterator and string methods.   - [development-policy](https://github.com/oxc-project/oxc/blob/64e4de78619eb9fc1e41d9595d7f209546b6ccb6/crates/oxc_linter/src/rules/eslint/max_lines.rs#L102-L106)

before:
```rs
if ignore_options.urls && URL_REGEXP.is_match(text_to_measure) {
    return true;
}
```
now:
```rs
if ignore_options.urls
    && (text_to_measure.contains("http://") || text_to_measure.contains("https://"))
{
    return true;
}
```